nulog-inference-service/NulogTrain.py

```diff
@@ -61,33 +84,47 @@ def s3_setup(s3_client):
             s3_client.create_bucket(Bucket=S3_BUCKET)
     return True
 
-async def send_signal_to_nats(nw):
-    await nw.publish(
-        "gpu_trainingjob_status", b"JobEnd"
-    )  ## tells the GPU service that a training job done.
-
-    nulog_payload = {
-        "bucket": S3_BUCKET,
-        "bucket_files": {
-            "model_file": "nulog_model_latest.pt",
-            "vocab_file": "vocab.txt",
-        },
-    }
-    await nw.publish(
-        nats_subject="model_ready", payload_df=json.dumps(nulog_payload).encode()
-    )
-    logger.info(
-        "Published to model_ready Nats subject that new Nulog model is ready to be used for inferencing."
-    )
+
+async def send_signal_to_nats(nw, training_success):
+    # Function that will send signal to Nats subjects gpu_trainingjob_status and model_ready.
+    await nw.connect()
+    # Regardless of a successful training of Nulog model, send JobEnd message to Nats subject gpu_trainingjob_status to make GPU available again.
+    await nw.publish("gpu_trainingjob_status", b"JobEnd")
+
+    # If Nulog model has been successfully trained, send payload to model_ready Nats subject that new model is ready to be uploaded from Minio.
+    if training_success:
+        nulog_payload = {
+            "bucket": S3_BUCKET,
+            "bucket_files": {
+                "model_file": "nulog_model_latest.pt",
+                "vocab_file": "vocab.txt",
+            },
+        }
+        await nw.connect()
```

Comment:
does it help to have `await nw.connect()` twice in the same function?

Reply:
yeah, I noticed at times the connection was lost after sending the first payload to the gpu_trainingjob_status Nats subject so that's why I have another connection made. This is the hot fix.